src/cpp/continuous_batching/src/sampler.hpp

```diff
@@ -375,10 +384,13 @@ GroupBeamSearcher::GroupBeamSearcher(SequenceGroup::Ptr sequence_group)
 }
 
 void GroupBeamSearcher::select_next_tokens(const ov::Tensor& logits, SamplerOutput& sampler_output) {
+    OPENVINO_ASSERT(m_parameters.num_beams % m_parameters.num_beam_groups == 0,
+        "number of beams should be divisible by number of groups");
```

Comment:
as I see - it's not part of generation config validation, let's move it there and remove from sampler code.